### DISCUSSION TITLE
Facebook strategy not working with @nuxtjs/auth and @nuxtjs/auth-next

### DISCUSSION THREAD

**Discussion by learntheropes** [nuxt-community/auth-module#1496] [Support Requests]
### Version
module: **4.9.1** and **5.0.0-1637745161.ea53f98**
nuxt: 2.15.8

#### [mode](https://nuxtjs.org/api/configuration-mode): 
 - [X ] universal
 - [ ] spa

### Nuxt configuration
**4.9.1** and **5.0.0-1637745161.ea53f98**
```
  auth: {
    redirect: {
      login: '/',
      logout: '/',
      callback: '/cb/',
      home: '/account/'
    },
    strategies: {
      facebook: {
        client_id: process.env.FACEBOOK_APP_ID,
        userinfo_endpoint: 'https://graph.facebook.com/v2.12/me?fields=id',
        scope: ['public_profile']
      },
    }
  },
```

## Reproduction

For **4.9.1**
git clone https://github.com/giovannilaperna/nuxt-facebook-login.git
cd nuxt-facebook-login
npm install
nuxt

For **5.0.0-1637745161.ea53f98**
git clone https://github.com/giovannilaperna/nuxt-next-facebook-login.git
cd nuxt-next-facebook-login
npm install
nuxt

#### What is expected?
Login using facebook and redirected to /account/ path

#### What is actually happening?

With **4.9.1**
Works locally
Doesn't work when deployed. auth.strategy cookies always set to local. See https://nuxt-facebook-login.netlify.app/

With **5.0.0-1637745161.ea53f98**
Doesn't work locally neither deployed.
In both case get facebook page with Sorry, something went wrong. See https://nuxt-next-facebook-login.netlify.app

#### Steps to reproduce
git clone the linked repos and run nuxt

## Checklist
* [X ]  I have tested with the latest Nuxt version and the issue still occurs
* [X ]  I have tested with the latest module version and the issue still occurs
* [ X]  I have searched the issue tracker and this issue hasn't been reported yet

**Comment by toniengelhardt** [comment]
Hey @giovannilaperna, I'm piping the request through a Django backend with allauth, which then in turn sends the request to Facebook. So the config for your case should be different, but here is what works for me:

```js
facebook: {
  clientId: process.env.FACEBOOK_CLIENT_ID,
  responseType: 'code',
  endpoints: {
    token: process.env.BACKEND_URL + '/auth/social/facebook/',
    userInfo: process.env.API_URL + '/user/',
  },
  token: {
    // By default the token type is Bearer, but simple_jwt is configured
    // to accept only JWT tokens.
    type: 'JWT',
  },
},
```

Note that it has `responseType: 'code'` and the endpoints for `token` and `userInfo` are nested inside a `endpoints` key. I think in your case you can omit the token type setting (but better to check what you get back from FB), but you might have to add `responseType`, nest `userInfo` inside `endpoints` and potentially specify the FB URL for the token.

Hope that helps.

**Comment by learntheropes** [comment]
@toniengelhardt my configuration works correctly in SPA mode. Strange that it doesn't work in universal mode.
However I had to switch to Auth0 strategy for other reason and it works correctly. Will test your suggestion when I have time just to confirm that there is a solution to the issue.  Thanks for your help
